File: app-kotlin/src/main/java/fudi/freddy/mokitosample/store/SharePreferenceContact.java
package fudi.freddy.mokitosample.store;

import fudi.freddy.mokitosample.model.Contact;
import com.romellfudi.sharepreference.SharePreferenced;

/**
 * @autor Romell Domínguez
 * @date 26/02/17
 * @version 1.0
 */


public class SharePreferenceContact implements StoreContract<Contact> {
    @Override
    public void save(Contact object) {
        SharePreferenced.save(object,Contact.class);
    }

    @Override
    public Contact load(String mail) {
        return SharePreferenced.load(Contact.class);
    }
}

File: app-kotlin/src/main/java/fudi/freddy/mokitosample/store/SharePreferenceContact.kt
package fudi.freddy.mokitosample.store

import com.romellfudi.sharepreference.SharePreferenced
import fudi.freddy.mokitosample.model.Contact

/**
 * @autor Romell Domínguez
 * @date 7/25/20
 * @version 1.0
 */
class SharePreferenceContact : StoreContract<Contact?> {
    override fun save(`object`: Contact?) {
        SharePreferenced.save(`object`, Contact::class.java)
    }

    override fun load(mail: String): Contact? {
        return SharePreferenced.load(Contact::class.java)
    }
}